.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateParser.cs

```diff
@@ -257,13 +264,13 @@ private DateTimeResolutionResult ParseImplicitDate(string text, DateObject refer
             if (match.Success)
             {
                 int day = 0, month = referenceDate.Month, year = referenceDate.Year;
-                var dayStr = match.Groups["day"].Value.ToLower();
+                var dayStr = match.Groups["DayOfMonth"].Value.ToLower();
 
                 // create a extract result which content ordinal string of text
                 ExtractResult er = new ExtractResult();
                 er.Text = dayStr;
-                er.Start = match.Groups["day"].Index;
-                er.Length = match.Groups["day"].Length;
+                er.Start = match.Groups["DayOfMonth"].Index;
```

Comment:
Shouldn't this be a constants somewhere as it's related to some resource/data definition?

Reply:
now, it will be single number (27), ordinary number (27th), number in text (twenty seven), ordinary number in text (twenty seventh). The resource/data definition doesn't contain all of them.  Therefore, we use the number parser to get the value. Should we create a task to config the 31*4 numbers to the regex as well as the resource/date definition?

Reply:
Let's not change this now. But please open an issue to analyze this case again soon.